docs/reference/remap.cue

```diff
@@ -52,17 +51,589 @@ remap: {
+
+				You can also use line collapsing via semicolon in [control flow
+				statements](#control-flow).
+				"""#
+		}
+
+		"Functions": {
+			href: "functions"
+
+			description: """
+				In TRL, functions can take inputs (or no input) and return either a value or, in the
+				case of some functions, an error.
+				"""
+
+			examples: [
+				"parse_json(.message)",
+				"assert(.status_code == 500)",
+				#"ip_subnet(.address, "255.255.255.0")"#,
+				".request_id = uuid_v4()",
+			]
+		}
+
+		"Control flow": {
+			href: "control-flow"
+
+			description: """
+				TRL supports control flow operations using `if`, `else if`, and `else`. These can
+				be called on any expression that returns a Boolean. Here's a generic example of the
+				syntax:
+
+				```
+				if (condition) {
+					...
+				} else if (other_condition) {
+					...
+				} else {
+					...
+				}
+				```
+
+				Any number of expressions can be combined inside of a block if they're separated by
+				a semicolon (`;`), provided that the last expression resolves to a Boolean:
+
+				```
+				if ($keyword = "sesame"; .password == $keyword) {
+					.entry = true
+				}
+				```
+				"""
+
+			examples: [
+				"""
+					if (match("this does contain foo", /(foo|bar)/)) {
+						.contains_foo = true
+					} else {
+						.does_not_contain_foo = true
+					}
+					""",
+			]
+		}
+
+		"Assignment": {
+			href: "assignment"
+
+			description: """
+				You can assign values to object fields or [variables](#variables) using a single
+				equals sign (`=`). Some examples:
+
+				* `.is_success = (.code > 200) && (.code <= 299)`
+				* `$pattern = /(foo|bar)/g`
+				* `. = parse_json(.)`
+				* `.request.id = uuid_v4()`
+
+				In TRL, `=` represents assignment, while `==` is a [comparison
+				operator](#operators), as in many programming languages.
+
+				If you assign a value to an object field that doesn't already exist, the field is
+				created; if the field does exist, the value is re-assigned.
+				"""
+
+			examples: [
+				".request_id = uuid_v4()",
+				"$average = .total / .number",
+				".partition_id = .status_code",
+				".is_server_error = .status_code == 500",
+			]
+		}
+
+		"Variables": {
+			href: "variables"
+
+			description: """
+				You can assign values to variables in TRL. Variables in TRL are prefixed with a `$`
+				and their names need to be [snake case](\(urls.snake_case)), as in `$myvar`,
+				`$my_var`, `$this_is_my_variable`, etc. Here's an example usage of a variable:
+
+				```
+				$log_level = "critical"
+				.log_level = $log_level
+				```
+
+				### Assignment using expressions
+
+				Because all TRL expressions return a value (by definition), you can assign using
+				expressions as well:
+
+				```
+				$is_success = .status_code == 200
+				$has_buzzword = contains(.message, "serverless")
+				```
+
+				### No regular expressions
+
+				All value types in TRL can be assigned to variables, with the important exception of
+				regular expressions.
```

Comment:
Regular expressions _can_ be assigned to variables, but you aren't allowed to:

* assign them to object paths
* return them as the final value of a program